### ISSUE TITLE
[Accessibility]: Code Insights Creation UI - Granularity description is never read

### ISSUE THREAD

**Issue by unclejustin** [sourcegraph/sourcegraph#41670]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/33494

### Problem description

![PixelSnap 2022-09-14 at 14 28 09](https://user-images.githubusercontent.com/1855233/190244985-4d5504db-faa3-41f3-933e-70be9760e46d.png)

The text "Show the past 3 months..." updates based on radio and spinner selection, but it is never read to the user.

### Expected behavior

This text should be read allowed when it changes.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.
